Formula/glibc.rb

```diff
@@ -95,41 +95,188 @@ class Glibc < Formula
 
   depends_on "binutils" => :build
   depends_on GawkRequirement => :build
-  depends_on "linux-headers@4.4" => :build
+  depends_on "linux-headers@5.15" => :build
   depends_on MakeRequirement => :build
   depends_on SedRequirement => :build
   depends_on BrewedGlibcNotOlderRequirement
   depends_on :linux
   depends_on LinuxKernelRequirement
 
-  # GCC 4.7 or later is required.
-  fails_with gcc: "4.6"
+  MINIMUM_MAKE_VERSION = "4.0".freeze
+  MINIMUM_GCC_VERSION = "6.2".freeze
+  INSTALLED_MAKE_VERSION = Utils.safe_popen_read("make", "--version").lines.first.split.third.freeze
+  INSTALLED_GCC_VERSION = Utils.safe_popen_read("gcc", "-dumpversion").freeze
 
-  def install
-    # Fix Error: `loc1@GLIBC_2.2.5' can't be versioned to common symbol 'loc1'
-    # See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=869717
-    # Fixed in glibc 2.24
-    inreplace "misc/regexp.c", /^(char \*loc[12s]);$/, "\\1 __attribute__ ((nocommon));"
+  if Version.new(INSTALLED_MAKE_VERSION) < MINIMUM_MAKE_VERSION
+    resource "make" do
+      url "https://ftp.gnu.org/gnu/make/make-4.3.tar.gz"
+      sha256 "e05fdde47c5f7ca45cb697e973894ff4f5d79e13b750ed57d7b66d8defc78e19"
+    end
+  end
+
+  if which("python3").blank?
```

Comment:
This (and same below) should be a `requirement` instead of using `which` here which will make the formula DSL output runtime dependent.

Reply:
If I understand how this works correctly, this means I would move the vendoring code into the Requirement instead of doing it here, because Requirement doesn’t return a Boolean value.  I will try to implement this as I understand it but may need some guidance.

Reply:
Yeh, the `Requirement` will make sure that the logic is only executed if the requirement fails.

Note, I'd really love to explore some other options here such as:
- always use the vendored code unconditionally so we don't have huge amounts of code that will never be used in CI
- use actual build-time formula dependencies (versioned if needed) instead of venturing
- explore if we can just avoid vendoring here entirely and push the requests onto the users who want to build from source

Reply:
> Yeh, the `Requirement` will make sure that the logic is only executed if the requirement fails.
> 
> Note, I'd really love to explore some other options here such as:
> 
> * always use the vendored code unconditionally so we don't have huge amounts of code that will never be used in CI
> * use actual build-time formula dependencies (versioned if needed) instead of venturing
> * explore if we can just avoid vendoring here entirely and push the requests onto the users who want to build from source

I would actually prefer to the use vendored code unconditionally if we're okay with that.  Only GCC takes a while to build and it's still not that bad, and it would make this a highly consistent, reproducible build.  This a formula that only gets installed once during setup and every 2 years after that, so increasing the build time somewhat is to me a worthwhile tradeoff for high reproducibility.  

I do not want to rely on external formulae (except `binutils` and `zlib` as we already do) or require the user to build the dependencies themselves, as these are both more prone to failure.  We have plenty of other formulae where we vendor dependencies and this works very nicely because we can guarantee everything will work in CI.

I'd like to move forward with the vendoring approach (preferably unconditionally) so we can merge this as soon as possible.  We will have ample opportunity to iterate on this later if needed but we have a simple solution that works here very well.

Reply:
What's the build time with and without vendoring? 

My ordered preference is:
- use actual build-time formula dependencies (versioned if needed)
- explore if we can just avoid vendoring here entirely and fail if the system configuration isn't what we need it to be (but we can make it this way when bottling)
- always use the vendored code unconditionally

so 👍🏻 to adjust to vendor unconditionally for now

Reply:
Okay that sounds good to me for now!